### PR TITLE
Use URLSession directly, and disable caching for DELETE request

### PR DESCRIPTION
**Pull request by zdnk**
Solves #19

### CODE REVIEW

**Comment by zdnk** [comment]
It passes the tests that I mention in issue #19 in my project.
https://github.com/zdnk/vapor-filesystem/tree/0c25a686238a368446ff511376fee9c824f9c8fb

File: Sources/S3/Extensions/S3+Private.swift [hunk 1]
```diff
@@ -1,28 +1,55 @@
...
+            let response = S3.convertFromFoundationResponse(httpResponse, data: data, on: container)
+            
+            promise.succeed(result: Response(http: response, using: container))
+        }).resume()
+        
+        return promise.futureResult
+    }
+    
+    /// Convert given response and data to HTTPResponse from Vapors HTTP package
+    static func convertFromFoundationResponse(_ httpResponse: HTTPURLResponse, data: Data?, on worker: Worker) -> HTTPResponse {
```
Comment (rafiki270) [inline Sources/S3/Extensions/S3+Private.swift]: `func convert(foundationResponse httpResponse: HTTPURLResponse, data: Data?, on worker: Worker) -> HTTPResponse`?

File: Sources/S3/Extensions/S3+Private.swift [hunk 2]
```diff
@@ -1,28 +1,55 @@
...
+        
+        return execute(request, on: container)
+    }
+    
+    /// Execute given request with URLSession.shared
+    func execute(_ request: URLRequest, on container: Container) -> Future<Response> {
+        let promise = container.eventLoop.newPromise(Response.self)
+        
+        URLSession.shared.dataTask(with: request, completionHandler: { (data, urlResponse, error) in
+            if let error = error {
```
Comment (rafiki270) [inline Sources/S3/Extensions/S3+Private.swift]: is there nay situation when we wouldn't hit the complete closure? (just asking) ... if there is'd have a future leak and NIO would crash

File: Sources/S3/Extensions/S3+Private.swift [hunk 3]
```diff
@@ -1,28 +1,55 @@
...
     /// Make an S3 request
-    func make(request url: URL, method: HTTPMethod, headers: HTTPHeaders, data: Data? = nil, on container: Container) throws -> Future<Response> {
-        let client = try container.make(Client.self)
-        let request = Request(using: container)
-        request.http.method = method
-        request.http.headers = headers
+    func make(request url: URL, method: HTTPMethod, headers: HTTPHeaders, data: Data? = nil, cachePolicy: URLRequest.CachePolicy = .useProtocolCachePolicy, on container: Container) throws -> Future<Response> {
+        var request = URLRequest(url: url, cachePolicy: cachePolicy)
+        request.httpMethod = method.string
+        request.httpBody = data ?? Data()
```
Comment (rafiki270) [inline Sources/S3/Extensions/S3+Private.swift]: can we just skip setting body altogether so we don't have to create an empty object?

File: Sources/S3/Extensions/S3+Private.swift [hunk 4]
```diff
@@ -1,28 +1,55 @@
...
+    
+    /// Convert given response and data to HTTPResponse from Vapors HTTP package
+    static func convertFromFoundationResponse(_ httpResponse: HTTPURLResponse, data: Data?, on worker: Worker) -> HTTPResponse {
+        var res = HTTPResponse(status: .init(statusCode: httpResponse.statusCode))
         if let data = data {
-            request.http.body = HTTPBody(data: data)
+            res.body = HTTPBody(data: data)
+        }
+        for (key, value) in httpResponse.allHeaderFields {
+            res.headers.replaceOrAdd(name: "\(key)", value: "\(value)")
```
Comment (rafiki270) [inline Sources/S3/Extensions/S3+Private.swift]:
```suggestion
            res.headers.replaceOrAdd(name: String(key), value: String(value))
```
is, I would say, a wee bit more readable

File: Sources/S3/Extensions/S3+Private.swift [hunk 5]
```diff
@@ -1,28 +1,55 @@
...
+            
+            promise.succeed(result: Response(http: response, using: container))
+        }).resume()
+        
+        return promise.futureResult
+    }
+    
+    /// Convert given response and data to HTTPResponse from Vapors HTTP package
+    static func convertFromFoundationResponse(_ httpResponse: HTTPURLResponse, data: Data?, on worker: Worker) -> HTTPResponse {
+        var res = HTTPResponse(status: .init(statusCode: httpResponse.statusCode))
```
Comment (rafiki270) [inline Sources/S3/Extensions/S3+Private.swift]: Could you please rename `res` to `response`?

File: Sources/S3/Extensions/S3+Private.swift (hunk 2 above)
Comment (zdnk) [reply Sources/S3/Extensions/S3+Private.swift]: Why would it crash? What leak?
Well it is Foundation and it is the only closure I can pass in the call so I suppose it is called every time :)

**Comment by zdnk** [comment]
I made the requested changes. Sorry for those smells, I just copied almost everything from `FoundationClient`.

File: Sources/S3/Extensions/S3+Private.swift (hunk 2 above)
Comment (rafiki270) [reply Sources/S3/Extensions/S3+Private.swift]: I am just asking ... NIO is super strict about having their futures fulfilled and the whole thing will crash on purpose if they aint

Comment (zdnk) [reply Sources/S3/Extensions/S3+Private.swift]: Well, the `FoundationClient` is doing the same so I hope it is ok.

**Comment by rafiki270** [comment]
Please put the header back ... I know some people don't like them but they are all across our code ... plus it has been removed from just one file only leaving the rest of the codebase inconsistent

**Comment by rafiki270** [comment]
also, would you mind contacting me on discord? @rafiki270

**Comment by zdnk** [comment]
> also, would you mind contacting me on discord? @rafiki270

I did :)

**Pull request merged**